contracts/fuses/tac/TacStakingFuse.sol

```diff
@@ -0,0 +1,273 @@
+import {TacStakingStorageLib} from "./TacStakingStorageLib.sol";
+import {TacValidatorAddressConverter} from "./TacValidatorAddressConverter.sol";
+
+/// @notice Struct to represent the data for entering the fuse - delegate action
+/// @dev validatorAddresses - the validator addresses
+/// @dev wTacAmounts - the amounts of TAC to delegate
+struct TacStakingFuseEnterData {
+    string[] validatorAddresses;
+    uint256[] wTacAmounts;
+}
+
+/// @notice Struct to represent the data for exiting the fuse - undelegate action
+/// @dev validatorAddresses - the array of validator addresses
+/// @dev wTacAmounts - the array of amounts of wTAC to unstake
+struct TacStakingFuseExitData {
+    string[] validatorAddresses;
+    uint256[] wTacAmounts;
+}
+
+/// @notice Struct to represent the data for redelegate action
+/// @dev validatorSrcAddresses - the source validator addresses
+/// @dev validatorDstAddresses - the destination validator addresses
+/// @dev wTacAmounts - the amounts of TAC to redelegate
+struct TacStakingFuseRedelegateData {
+    string[] validatorSrcAddresses;
+    string[] validatorDstAddresses;
+    uint256[] wTacAmounts;
+}
+
+contract TacStakingFuse is IFuseCommon, IFuseInstantWithdraw {
+    using SafeERC20 for IERC20;
+
+    error TacStakingFuseInvalidDelegatorAddress();
+    error TacStakingFuseSubstrateNotGranted(string validator);
+    error TacStakingFuseDelegatorAlreadyCreated();
+    error TacStakingFuseArrayLengthMismatch();
+    error TacStakingFuseEmptyArray();
+    error TacStakingFuseInsufficientBalance();
+
+    event TacStakingFuseEnter(address version, string[] validatorAddresses, uint256[] wTacAmounts);
+    event TacStakingFuseExit(address version, string[] validatorAddresses, uint256[] wTacAmounts);
+    event TacStakingFuseRedelegate(
+        address version,
+        string[] validatorSrcAddresses,
+        string[] validatorDstAddresses,
+        uint256[] wTacAmounts
+    );
+    event TacStakingFuseInstantWithdraw(address version, uint256 amount);
+    event TacStakingDelegatorCreated(address delegator, address plasmaVault, address wTAC, address staking);
+
+    address public immutable VERSION;
+    uint256 public immutable MARKET_ID;
+    address public immutable W_TAC;
+    address public immutable STAKING;
+
+    constructor(uint256 marketId_, address wTAC_, address staking_) {
+        if (wTAC_ == address(0)) {
+            revert TacStakingFuseInvalidDelegatorAddress();
+        }
+        if (staking_ == address(0)) {
+            revert TacStakingFuseInvalidDelegatorAddress();
+        }
+        VERSION = address(this);
+        MARKET_ID = marketId_;
+        W_TAC = wTAC_;
+        STAKING = staking_;
+    }
+
+    /// @notice Creates a new TacStakingDelegator and stores its address in storage
+    /// @dev Only callable by alpha role
+    /// @dev Creates a new TacStakingDelegator and stores its address in storage, can be called only once
+    function createDelegator() external {
```

Comment:
Couldn’t this be automated in a way that if it doesn’t exist, it will be created automatically?